src/phases/tpch/denormalized/Q2.yml

```diff
@@ -18,19 +18,17 @@ TPCHDenormalizedQuery2:
     OperationName: RunCommand
     OperationCommand:
       explain:
-        aggregate: partsupp
+        aggregate: part
         pipeline:
           [
-            {$lookup: {from: "part", as: "part", localField: "ps_partkey", foreignField: "p_partkey", pipeline: [
-              {$match: {$and: [{p_type: {$regex: *query2Type, "$options": "si"}}, {p_size: {$eq: *query2Size}}]}}]}},
-            {$unwind: "$part"},
-            {$lookup: {from: "supplier", localField: "ps_suppkey", foreignField: "s_suppkey", as: "supplier", pipeline: [
-              {$match: {"nation.region.r_name": *query2Region}}]}},
+            {$match: {$and: [{p_type: {$regex: *query2Type, $options: "si"}}, {p_size: {$eq: *query2Size}}]}},
```

Comment:
Good call. Pushing down `$match` looks a good optimization to speed up the query. You may want to update the design doc as well :)

Reply:
Done!